lib/FileUploader/utils/uploadFile.js

```diff
@@ -0,0 +1,57 @@
+import {
+  forEach,
+  noop,
+} from 'lodash';
+
+import { createOkapiHeaders } from '../../utils';
+
+const addXHRHeaders = (xhr, headers) => {
+  forEach(headers, (value, key) => xhr.setRequestHeader(key, value));
+};
+
+export async function uploadFile({
+  xhr,
+  file,
+  url,
+  okapi,
+  onFileUploadProgress = noop,
+}) {
+  return new Promise((resolve, reject) => {
+    try {
+      xhr.open('POST', url);
+
+      addXHRHeaders(xhr, {
+        ...createOkapiHeaders(okapi),
+        'Content-Type': 'application/octet-stream',
+      });
+
+      xhr.upload.onprogress = event => onFileUploadProgress(file, event);
+
+      xhr.onreadystatechange = () => {
+        const {
+          status,
+          readyState,
+          responseText,
+        } = xhr;
+
+        if (readyState !== 4) return;
+
+        try {
+          const parsedResponse = JSON.parse(responseText);
+
+          if (status === 200) {
```

Comment:
```suggestion
         status === 200 ? resolve(parsedResponse) : reject(parsedResponse);
```

Reply:
I prefer to use this notation only for the case when we assign the result to the variable. In other cases `if` is a good approach for it I think. Moreover, it will allow adding code later in the branches easily.